Fix linting issues in photometry

For all of [these issues](https://github.com/feder-observatory/stellarphot/issues?q=is%3Aopen+is%3Aissue+label%3Alinting), first delete the appropriate line in `pyproject.toml` that excludes the subpackage from checks. for `photometry` you would delete [this line](https://github.com/feder-observatory/stellarphot/blob/main/pyproject.toml#L123):

```toml
"stellarphot/photometry/*" = ["F"]
```

After that, run `ruff` either with `tox -e lint` or `ruff .`. The errors you get should be the ones -- fix them! If it is an unused function argument make sure to search the project for calls to that function to update the arguments there too.

- [ ] stellarphot/photometry/tests/test_photometry.py:526:9: F841 Local variable `expected_deviation` is assigned to but never used
- [ ] stellarphot/photometry/tests/test_photometry.py:578:13: F841 Local variable `phot_data` is assigned to but never used
- [ ] stellarphot/photometry/tests/test_photometry.py:637:13: F841 Local variable `phot_data` is assigned to but never used

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -123,7 +123,6 @@ select = [
 
 # Ignore pyflakes and ARGS issues in each subpackage until they are fixed
 "stellarphot/gui_tools/*" = ["F", "ARG"]
-"stellarphot/photometry/*" = ["F"]
 "stellarphot/utils/*" = ["F", "ARG"]
 
 [tool.pytest.ini_options]

--- a/stellarphot/photometry/profiles.py
+++ b/stellarphot/photometry/profiles.py
@@ -124,7 +124,7 @@ def find_center(image, center_guess, cutout_size=30, max_iters=10, match_limit=3
 
     # Is we hit the max number of iterations, raise an error
     if max_iters > 1 and cnt > max_iters:
-        raise RuntimeError(f"Centroid finding did not converge")
+        raise RuntimeError("Centroid finding did not converge")
 
     # Get the final centroid position by fitting a gaussian
     # We may not have converged on a star, so capture any warning about a

--- a/stellarphot/photometry/tests/test_photometry.py
+++ b/stellarphot/photometry/tests/test_photometry.py
@@ -544,9 +544,7 @@ def test_photometry_on_directory():
 
         # Here we just check whether any difference is consistent with
         # less than the expected one sigma deviation.
-        assert (
-            np.abs(expected_flux - obs_avg_net_cnts) < np.pi * aperture**2 * noise_dev
-        )
+        assert np.abs(expected_flux - obs_avg_net_cnts) < expected_deviation
 
 
 def test_photometry_on_directory_with_no_ra_dec():
@@ -587,7 +585,7 @@ def test_photometry_on_directory_with_no_ra_dec():
         found_sources.drop_ra_dec()
 
         with pytest.raises(ValueError):
-            phot_data = multi_image_photometry(
+            multi_image_photometry(
                 temp_dir,
                 object_name,
                 found_sources,
@@ -645,7 +643,7 @@ def test_photometry_on_directory_with_bad_fits():
 
         # Since none of the images will be valid, it should raise a RuntimeError
         with pytest.raises(RuntimeError):
-            phot_data = multi_image_photometry(
+            multi_image_photometry(
                 temp_dir,
                 object_name,
                 found_sources,

--- a/stellarphot/photometry/tests/test_profiles.py
+++ b/stellarphot/photometry/tests/test_profiles.py
@@ -3,7 +3,6 @@ import numpy as np
 import pytest
 
 from astropy.table import Table
-from astropy.utils.exceptions import AstropyUserWarning
 
 from photutils.datasets import make_gaussian_sources_image, make_noise_image
 
@@ -72,7 +71,7 @@ def test_find_center_no_star():
     )
 
     with pytest.raises(RuntimeError, match="Centroid did not converge on a star"):
-        cen = find_center(image + noise, [50, 200], max_iters=10)
+        find_center(image + noise, [50, 200], max_iters=10)
 
 
 def test_radial_profile():

